### DISCUSSION TITLE
[General] <Title>ZK Sync In Memory Node Specs

### DISCUSSION THREAD

**Discussion by XOFEARTH** [zkSync-Community-Hub/zksync-developers#372] [General]
### Environment

Testnet

### Provide a brief description of the functionality you're trying to implement.

Currently interested in starting to get into testing the network via the In Memory node deployment. 

### What is the specific issue or error you're encountering?

I need to know what specs are needed for the device running the node? Can not find any documentations on the actual requirements. 

### Can you share the error messages or logs you're receiving, if any?

*

### Have you made any recent changes to the contract before encountering this issue?

*

### Are there any external libraries or contracts that your contract interacts with?

*

### Can you provide the relevant portions of your contract code where the issue is occurring?

*

### Have you tried to isolate the problem, and if so, what were the results?

*

### What steps have you already taken to try to resolve the issue?

*

**Comment by knowalosbra** [comment]
To run an in-memory node for testing purposes, you typically need a machine with sufficient RAM and CPU resources. While there's no specific documentation detailing the exact specifications required, here are some general recommendations:

RAM: At least 8GB of RAM is recommended to ensure smooth operation of the in-memory node.
CPU: A modern multi-core processor (e.g., Intel Core i5 or Ryzen 5) should suffice for running the node efficiently.
Storage: Since the node operates in-memory, storage requirements are minimal. However, having sufficient disk space for storing temporary files and logs is still important.
Keep in mind that these are general recommendations, and the actual resource requirements may vary based on factors such as the size of the network, the complexity of the contracts being tested, and the volume of transactions.

It's also advisable to allocate additional resources if you plan to run other applications alongside the in-memory node to prevent performance degradation.

**Comment by kaycogrisitbran** [comment]
For running an In Memory node deployment on the zkSync Testnet, you would typically need a device with sufficient RAM and CPU resources. Unfortunately, specific requirements are not documented, but a modern machine with at least 8GB of RAM and a multi-core CPU should suffice for testing purposes.

**Comment by dutterbutter** [comment]
@XOFEARTH did you try running era-test-node on your machine first? The hardware requirements are generally low.

**Comment by shannonzimmons** [comment]
The user is interested in testing the Testnet network via In Memory node deployment, but is unsure of the specifications required for the device running the node. They are seeking documentation on the actual requirements but have not found any. No error messages or logs are provided. No recent changes to the contract have been made, and no external libraries or contracts are being used. The user has not provided any relevant portions of their contract code, nor have they attempted to isolate the problem. They are seeking guidance on the specifications required for running an In Memory node on Testnet.
